custom/staking/hook.go

```diff
@@ -0,0 +1,92 @@
+package staking
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+const (
+	ColumbusChainID = "columbus-5"
+)
+
+var _ stakingtypes.StakingHooks = &TerraStakingHooks{}
+
+// TerraStakingHooks implements staking hooks to enforce validator power limit
+type TerraStakingHooks struct {
+	sk stakingkeeper.Keeper
+}
+
+func NewTerraStakingHooks(sk stakingkeeper.Keeper) *TerraStakingHooks {
+	return &TerraStakingHooks{sk: sk}
+}
+
+// Implement required staking hooks interface methods
+func (h TerraStakingHooks) BeforeDelegationCreated(_ sdk.Context, _ sdk.AccAddress, _ sdk.ValAddress) error {
+	return nil
+}
+
+func (h TerraStakingHooks) BeforeDelegationSharesModified(_ sdk.Context, _ sdk.AccAddress, _ sdk.ValAddress) error {
+	return nil
+}
+
+// Other required hook methods with empty implementations
+func (h TerraStakingHooks) AfterDelegationModified(ctx sdk.Context, _ sdk.AccAddress, valAddr sdk.ValAddress) error {
```

Comment:
The reason why i don't use those hooks is because the hooks  only provide us with this context `(ctx sdk.Context, _ sdk.AccAddress, valAddr sdk.ValAddress)`, which is not-sufficient to perform the pre-check behavior like in the current forks.  (we need the `types.Msg...` to get the required information)